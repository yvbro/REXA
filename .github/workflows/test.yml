name: tests

on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]

jobs:
  build:
    runs-on: ubuntu-latest
    services:
      postgres:
        image: postgres:12-alpine
        env:
          POSTGRES_USER: rexaAdmin
          POSTGRES_PASSWORD: rexaAdmin
          POSTGRES_DB: rexa
          POSTGRES_HOST: localhost
        ports:
          - 5442:5432
        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5

    steps:
      - uses: actions/checkout@v2
      - name: Set up JDK 11
        uses: actions/setup-java@v1
        with:
          java-version: 11
      - name: Init db
        run: mvn flyway:migrate
      - name: Unit test and integration test
        run: mvn -B clean verify --file pom.xml