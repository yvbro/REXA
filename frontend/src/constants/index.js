export const API_BASE_URL = 'http://localhost:8080';
export const ACCESS_TOKEN = 'accessToken';

export const OAUTH2_REDIRECT_URI = 'http://localhost:8080/oauth2/redirect'

export const GOOGLE_AUTH_URL = API_BASE_URL + '/oauth2/authorization/google?redirect_uri=' + OAUTH2_REDIRECT_URI;
