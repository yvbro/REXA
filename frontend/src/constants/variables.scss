$navSize: 80px;
$headerTextSize: 30px;
$iconSize: 1.75em;
$spacingSize: 20px;
$borderColor: #e1e4e7;
