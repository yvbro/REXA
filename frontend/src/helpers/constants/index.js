export const API_BASE_URL = 'http://localhost:8080';
export const ACCESS_TOKEN = 'accessToken';
export const TOKEN_TYPE = 'Bearer';
export const EXPIRATION_DATE = 'expirationDate';

export const OAUTH2_REDIRECT_URI = 'http://localhost:8080/oauth2/redirect';
export const GOOGLE_AUTH_PROVIDER = 'google';

export const GOOGLE_AUTH_URL =
    API_BASE_URL +
    '/oauth2/authorize-client/google?redirect_uri=' +
    OAUTH2_REDIRECT_URI;

// eslint-disable-next-line
export const regexEmail = /^\w+([\.-]?\w+)+@\w+([\.:]?\w+)+(\.[a-zA-Z0-9]{2,3})+$/;

const REGEX_ONE_CAPITAL_LETTER = /(?:^|[^A-Z])[A-Z](?![A-Z])/;
const REGEX_ONE_NUMBER = /^.*[0-9]+.*$/;
const PASSWORD_LENGTH = 8;

export const ERROR_INVALID_PASSWORD =
    'The new password is invalid. Please follow the rules.';

export const isPasswordInvalid = (password) => {
    return (
        !password.match(REGEX_ONE_CAPITAL_LETTER) ||
        !password.match(REGEX_ONE_NUMBER) ||
        password.length < PASSWORD_LENGTH
    );
};
