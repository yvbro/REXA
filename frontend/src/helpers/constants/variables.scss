$theme-core: (
  'headerTextSize': 30px,
  'navSize': 64px,
);

$theme-size: (
  'borderRadius': 5px,
  'spacingSize': 20px,
  'iconSize': 1.75em,
);

$theme-colors: (
  'borderColor': #fff,
  'backgroundColor': #f0f2f5,
  'themeColor': #031831,
  'textColor': #30373e,
  'primaryButtonColor': #3f51b5,
  'primaryButtonDisabledColor': #4762a6,
  'secondaryButtonColor': #bd0f0f,
);

:export {
  @each $key, $value in $theme-core {
    #{$key}: #{$value};
  }

  @each $key, $value in $theme-size {
    #{$key}: #{$value};
  }

  @each $key, $value in $theme-colors {
    #{$key}: #{$value};
  }
}
