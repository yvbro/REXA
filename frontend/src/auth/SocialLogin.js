import React from 'react'

import {GOOGLE_AUTH_URL} from '../constants';
import googleLogo from '../assets/google-logo.png';

const SocialLogin = () => (
    <div className="social-login">
        <a className="btn btn-block social-btn google" href={GOOGLE_AUTH_URL}>
            <img src={googleLogo} alt="Google"/> Sign in with Google</a>
    </div>
);

export default SocialLogin;
