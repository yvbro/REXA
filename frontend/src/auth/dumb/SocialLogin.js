import React from 'react'

import {Button, makeStyles, Icon} from "@material-ui/core";

import {GOOGLE_AUTH_URL} from '../../constants';
import googleLogo from '../../assets/google-logo.png';

const useStyles = makeStyles((theme) => ({
    button: {
        margin: theme.spacing(1),
        width: 350,
    },
    imageIcon: {
        height: '100%'
    },
    iconRoot: {
        textAlign: 'center'
    }
}));

const GoogleIcon = () => {
    const classes = useStyles();
    return (
        <Icon classes={{root: classes.iconRoot}}>
            <img className={classes.imageIcon} src={googleLogo} alt="Google"/>
        </Icon>)
};

const SocialLogin = () => {
    const classes = useStyles();

    return (
        <Button
            variant="outlined"
            color="primary"
            className={classes.button}
            href={GOOGLE_AUTH_URL}
            startIcon={<GoogleIcon/>}
        >
            Sign in with Google
        </Button>
    )
};

export default SocialLogin;
