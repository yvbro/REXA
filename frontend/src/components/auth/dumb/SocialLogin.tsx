import React from 'react';

import { Button, makeStyles, Icon } from '@material-ui/core';

import { GOOGLE_AUTH_URL } from '../../../helpers/constants';
import googleLogo from '../../../assets/google-logo.png';
import classes from './auth.module.scss';

const useStyles = makeStyles((theme) => ({
    button: {
        margin: theme.spacing(1),
        width: 350,
    },
}));

export function GoogleIcon() {
    return (
        <Icon classes={{ root: classes.iconRoot }}>
            <img className={classes.imageIcon} src={googleLogo} alt="Google" />
        </Icon>
    );
}

function SocialLogin() {
    const style = useStyles();

    return (
        <Button
            variant="outlined"
            color="primary"
            className={style.button}
            href={GOOGLE_AUTH_URL}
            startIcon={<GoogleIcon />}
        >
            Sign in with Google
        </Button>
    );
}

export default SocialLogin;
