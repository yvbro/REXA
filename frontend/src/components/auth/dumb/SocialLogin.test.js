import React from 'react';
import { renderWithStore, cleanup } from '../../../helpers/test/test-utils';
import '@testing-library/jest-dom/extend-expect';

import SocialLogin from './SocialLogin';

import { GOOGLE_AUTH_URL } from '../../../helpers/constants/index';

describe('The SocialLogin component', () => {
    afterEach(cleanup);

    it('should take a snapshot', () => {
        const { asFragment } = renderWithStore(<SocialLogin />);

        expect(asFragment(<SocialLogin />)).toMatchSnapshot();
    });

    it('should display a link as a button with the text "Sign in with Google"', () => {
        const { getByRole } = renderWithStore(<SocialLogin />);

        expect(getByRole('link')).toHaveTextContent('Sign in with Google');
        expect(getByRole('link')).toHaveAttribute('href', GOOGLE_AUTH_URL);
    });
});
