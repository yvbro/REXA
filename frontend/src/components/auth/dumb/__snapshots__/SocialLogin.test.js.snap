// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The SocialLogin component should take a snapshot 1`] = `
<DocumentFragment>
  <a
    aria-disabled="false"
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-1 MuiButton-outlinedPrimary"
    href="http://localhost:8080/oauth2/authorize-client/google?redirect_uri=http://localhost:8080/oauth2/redirect"
    tabindex="0"
  >
    <span
      class="MuiButton-label"
    >
      <span
        class="MuiButton-startIcon MuiButton-iconSizeMedium"
      >
        <span
          aria-hidden="true"
          class="material-icons MuiIcon-root iconRoot"
        >
          <img
            alt="Google"
            class="imageIcon"
            src="google-logo.png"
          />
        </span>
      </span>
      Sign in with Google
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </a>
</DocumentFragment>
`;
